server/metadata/tenant.go

```diff
@@ -166,6 +166,39 @@ func (m *TenantManager) CreateOrGetTenant(ctx context.Context, txMgr *transactio
 	return m.createOrGetTenantInternal(ctx, tx, namespace)
 }
 
+// CreateTenant is a thread safe implementation of creating a new tenant. It returns the error if it already exists.
+func (m *TenantManager) CreateTenant(ctx context.Context, txMgr *transaction.Manager, namespace Namespace) (tenant *Tenant, err error) {
+	m.Lock()
+	defer m.Unlock()
+
+	var ok bool
+	if tenant, ok = m.tenants[namespace.Name()]; ok {
+		if tenant != nil {
+			return tenant, api.Errorf(api.Code_CONFLICT, "namespace already exists with id '%s'", fmt.Sprint(tenant.namespace.Id()))
+		}
+	}
+
+	tx, e := txMgr.StartTx(ctx)
+	if ulog.E(e) {
+		return nil, e
+	}
+
+	defer func() {
+		if err == nil {
+			if err = tx.Commit(ctx); err == nil {
+				// commit succeed, so we can safely cache it now, for other workers it may happen as part of the
+				// first call in query lifecycle
+				m.tenants[namespace.Name()] = tenant
```

Comment:
Okay, we need to cache then keep the transaction logic inside this method, ignore my comment below.

Reply:
Actually, we don't need to cache it. Let's keep it simple. CreateTenant is only responsible for creating a tenant if it doesn't exist. Then you also don't need transaction logic inside this method and can move the transaction logic outside.

Reply:
ok moved the caching part out of this method.